src/ui.rs

```diff
@@ -151,19 +225,22 @@ impl Ui {
         let data = text.as_bytes();
         match &mut self.output {
             UiOutput::Terminal { stdout, .. } => stdout.write_all(data),
+            UiOutput::Paged { child } => child.stdin.take().unwrap().write_all(data),
```

Comment:
`child.stdin.take()` will remove the value from `child`, so this can only be called once, I think. I meant that you can store both `Child` and `ChildStdin` as separate fields in `UiOutput::Paged` if that works.

Reply:
> I meant that you can store both Child and ChildStdin as separate fields in UiOutput::Paged if that works

If it possible, it's quite beyond my current Rust skills 😛

Reply:
I pushed https://github.com/martinvonz/jj/commit/55dc237309ad6d634428e9b89e8deb7723241a3d to show what I meant :)

Reply:
Why does it look so easy when you do it??

Reply:
This sounds absurd, but when I make this change, `less` no longer recognizes the end of input when I scroll to the end. I have no idea why this might be. Let me ask for advice.

Reply:
This is because you're no longer dropping `ChildStdin` at any point before `less` exits. See further discussion below.